R/helpers.R

```diff
@@ -19,7 +19,12 @@
 #'
 #' @export
 keep_units <- function(FUN, x, ..., unit=units(x)) {
-  set_units(do.call(FUN, list(x, ...)), unit, mode="standard")
+  if (inherits(try(unit, silent = TRUE), "symbolic_units")) {
+    set_units(do.call(FUN, list(x, ...)), unit, mode = "standard")
+  } else {
+    warning("`x` does not have units.")
```

Comment:
Maybe "wrong `unit` specification" or something similar? Because the user may have defined `unit` explicitly and still get it wrong.